### PR TITLE
Feature/SOC5K-231: passwordcomplexity

### PR DESCRIPTION
**Pull request by faisal-hameed**
CI : http://jenkins.aureacentral.com/job/5k-rewrite/job/Sococo/job/Sococo-CI/view/change-requests/job/PR-25/6/

### CODE REVIEW

**Comment by emil-cazacu-aurea** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

**Comment by emil-cazacu-aurea** [comment]
:white_check_mark: No issues identified in this pull request. 
    
**Good work!**
<details>
    <summary>T4W FTAR Statistics</summary>
    <br />
    CodeRead team introduced new quality measurement to evaluate the trailing 4 weeks First Time Acceptance Rate of Pull Requests, this PR increased your FTAR from 42.86 % to 50.00 %, good job, and for more quality improvements you can read about yellow findings.
	<br />
    -
    <details>
        <summary>More Details</summary>
        <br />
        <table>
            <tr>
                <td>Current FTAR</td>
                <td><b>50.00 %</b></td>
            </tr>
            <tr>
                <td>Total Reviewed Pull Requests for the last 4 weeks</td>
                <td>8</td>
            </tr>
            <tr>
                <td>Passed</td>
                <td>4</td>
            </tr>
            <tr>
                <td>Rejected</td>
                <td>4</td>
            </tr>
        </table>
        More details about how FTAR is calculated, refer to this page - <a
            href="https://confluence.devfactory.com/x/Cw8dGg">FTAR - First time acceptance rate</a>
    </details>

</details>

**Comment by emil-cazacu-aurea** [comment]
Hi, I'm Emil Cazacu from [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members) and I have just reviewed your PR.
<b>How likely are you to recommend the Code Review service to another developer?</b>

Not at all likely \[ [1](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/5k-sococo/pull/25&entry.1163746017=1) \] \[ [2](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/5k-sococo/pull/25&entry.1163746017=2) \] \[ [3](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/5k-sococo/pull/25&entry.1163746017=3) \] \[ [4](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/5k-sococo/pull/25&entry.1163746017=4) \] \[ [5](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/5k-sococo/pull/25&entry.1163746017=5) \] \[ [6](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/5k-sococo/pull/25&entry.1163746017=6) \] \[ [7](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/5k-sococo/pull/25&entry.1163746017=7) \] \[ [8](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/5k-sococo/pull/25&entry.1163746017=8) \] \[ [9](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/5k-sococo/pull/25&entry.1163746017=9) \] \[ [10](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/5k-sococo/pull/25&entry.1163746017=10) \] Very likely

> TIP: Press `Ctrl` (or `⌘` in a Mac) to open the survey in a new tab.

File: backend/lambda/src/model/response.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export namespace Response {
+	export interface Result {
+		success: boolean;
+	}
+
+	export function success(): Result {
+		return { success: true };
+	}
+}
```
Comment (xoadrian) [inline backend/lambda/src/model/response.ts]: please reformat this file to use 2 paces + configure the IDE to do it automatically

File: backend/lambda/src/service/organization.ts [hunk 2]
```diff
@@ -115,4 +115,27 @@ export namespace OrganizationService {
...
+  export async function setPasswordComplexity(
+    orgId: string,
+    length: number,
+    expire: number,
+    complexity: boolean
+  ): Promise<Organization> {
+    const org: Organization = await getOrganization(orgId);
+
+    const orgPrefs: any = org.doc.preferences || {};
+
```
Comment (xoadrian) [inline backend/lambda/src/service/organization.ts]: remove empty line between constants, add new line after them.

File: backend/lambda/src/service/organization.ts [hunk 3]
```diff
@@ -115,4 +115,27 @@ export namespace OrganizationService {
...
+    const pwdPrefs: any = {};
+    pwdPrefs.length = length || undefined;
+    pwdPrefs.expire = expire || undefined;
+    pwdPrefs.complexity = complexity || undefined;
+
```
Comment (xoadrian) [inline backend/lambda/src/service/organization.ts]: remove empty line

**Comment by faisal-hameed** [comment]
Closing outdated PR

**Pull request closed without merging**